syft/frameworks/torch/hook/hook.py

```diff
@@ -468,6 +473,62 @@ def overloaded_attr(self, *args, **kwargs):
 
         return overloaded_attr
 
+    def _hook_promise_tensor(hook_self):
+
+        methods_to_hook = hook_self.to_auto_overload[torch.Tensor]
+
+        def generate_method(method_name):
+            def method(self, *args, **kwargs):
+
+                arg_shapes = list([self._shape])
+                arg_ids = list([self.id])
+
+                # Convert scalar arguments to tensors to be able to use them with plans
+                args = list(args)
+                for ia in range(len(args)):
+                    if not isinstance(args[ia], (torch.Tensor, AbstractTensor)):
+                        args[ia] = torch.tensor(args[ia])
```

Comment:
I'm surprised that this is needed: you can usually use scalars with torch operation:
A cmd like `th.tensor([1.]) + 1` will (unless I'm wrong) be serialized and store the 1 as a scalar

Reply:
I understand it will make your next call to .shape fail, but you can test there if it has a shape

Reply:
It was to be able to use plans with scalar arguments.
I don't think it works to call a plan for which some arguments are Python integers, for instance.

Reply:
I think it depends on the use case: for NN stuff maybe it's not needed but for crypto protocols, it might be more often (I'm not a crypto expert though).
And the problem is that we need the arg to have an id so that we can find it back. Or we can just store the scalar value and use it instead of an id but this needs some changes

Reply:
I think you can "just store the scalar value" in the plan, it looks like it works from what I've seen

Reply:
I still think it's easier to change the scalars to tensors because this way, I can store and retrieve them from the worker when needed (when the promises are fulfilled). I can't see how to do that with scalars (but maybe it's just me :))